src/Altinn.App.Core/Features/Validation/Default/SigningTaskValidator.cs

```diff
@@ -0,0 +1,159 @@
+        IDataClient dataClient,
+        IInstanceClient instanceClient,
+        IAppMetadata appMetadata,
+        ModelSerializationService modelSerialization
+    )
+    {
+        _logger = logger;
+        _processReader = processReader;
+        _signingService = signingService;
+        _dataClient = dataClient;
+        _instanceClient = instanceClient;
+        _appMetadata = appMetadata;
+        _modelSerialization = modelSerialization;
+    }
+
+    /// <summary>
+    /// We implement <see cref="ShouldRunForTask"/> instead.
+    /// </summary>
+    public string TaskId => "*";
+
+    /// <summary>
+    /// Only runs for tasks that are of type "signing".
+    /// </summary>
+    public bool ShouldRunForTask(string taskId)
+    {
+        AltinnTaskExtension? taskConfig;
+        try
+        {
+            taskConfig = _processReader.GetAltinnTaskExtension(taskId);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, $"Error while fetching signing configuration for task {taskId}");
+            return false;
+        }
+
+        AltinnSignatureConfiguration? signingConfiguration = taskConfig?.SignatureConfiguration;
+
+        return signingConfiguration?.RunDefaultValidator == true && taskConfig?.TaskType is "signing";
+    }
+
+    public bool NoIncrementalValidation => true;
+
+    /// <summary>
+    /// Should never be called because NoIncrementalValidation is true.
+    /// </summary>
+    public Task<bool> HasRelevantChanges(IInstanceDataAccessor dataAccessor, string taskId, DataElementChanges changes)
+    {
+        throw new NotImplementedException();
+    }
+
+    /// <inheritdoc />
+    /// <remarks>Validates that all required parties have signed the current task.</remarks>
+    public async Task<List<ValidationIssue>> Validate(
+        IInstanceDataAccessor dataAccessor,
+        string taskId,
+        string? language
+    )
+    {
+        AltinnSignatureConfiguration? signingConfiguration = (
+            _processReader.GetAltinnTaskExtension(taskId)?.SignatureConfiguration
+        );
+
+        if (signingConfiguration == null)
+        {
+            _logger.LogError($"No signing configuration found for task {taskId}");
+            return [];
+        }
+
+        var (getAppMetadataError, appMetadata) = await CatchError(_appMetadata.GetApplicationMetadata());
+        if (getAppMetadataError != null || appMetadata == null)
+        {
+            _logger.LogError(getAppMetadataError, "Error while fetching application metadata");
+            return [];
+        }
+
+        var cachedDataMutator = new InstanceDataUnitOfWork(
+            instance: dataAccessor.Instance,
+            _dataClient,
+            _instanceClient,
+            appMetadata,
+            _modelSerialization
+        );
+
+        var (getSigneeContextsError, signeeContexts) = await CatchError(
+            _signingService.GetSigneeContexts(cachedDataMutator, signingConfiguration)
```

Comment:
You should be able to get the signeeContexts from the `dataAccessor` instead of instantiating a `Mutator` in a validator.

Reply:
You would need to update the signatures in the signing service to use `IInstanceDataAccessor`

Reply:
The `GetSigneeContexts`-function also creates new contexts if data from different sources don't align, so it seems like it needs the mutator unfortunately :/

Reply:
Nm, I was wrong 🙃